src/Cake.Prca/IssueFilterer.cs

```diff
@@ -0,0 +1,178 @@
+
+        /// <summary>
+        /// Filters all issues which should not be logged.
+        /// </summary>
+        /// <param name="issues">Found issues.</param>
+        /// <param name="issueComments">List of existing comments on the pull request.</param>
+        /// <returns>List of filtered issues.</returns>
+        public IEnumerable<ICodeAnalysisIssue> FilterIssues(
+            IEnumerable<ICodeAnalysisIssue> issues,
+            IDictionary<ICodeAnalysisIssue, IEnumerable<IPrcaDiscussionComment>> issueComments)
+        {
+            // ReSharper disable once PossibleMultipleEnumeration
+            issues.NotNull(nameof(issues));
+            issueComments.NotNull(nameof(issueComments));
+
+            this.log.Verbose("Filtering issues before posting...");
+
+            // ReSharper disable once PossibleMultipleEnumeration
+            var codeAnalysisIssues = issues as IList<ICodeAnalysisIssue> ?? issues.ToList();
+
+            var result = this.FilterIssuesByPath(codeAnalysisIssues);
+            result = this.FilterPreExistingComments(result, issueComments);
+            result = this.FilterIssuesByNumber(result);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Checks if there's already a comment for an issue.
+        /// </summary>
+        /// <param name="issue">Issue to check.</param>
+        /// <param name="issueComments">List of existing comments.</param>
+        /// <returns>True if there are already comments for an issue.</returns>
+        private static bool IssueHasMatchingComments(
+            ICodeAnalysisIssue issue,
+            IDictionary<ICodeAnalysisIssue, IEnumerable<IPrcaDiscussionComment>> issueComments)
+        {
+            return
+                issueComments.ContainsKey(issue) &&
+                issueComments[issue].Any();
+        }
+
+        /// <summary>
+        /// Filters all issues affecting files which do not belong to files changed in this pull request.
+        /// </summary>
+        /// <param name="issues">List of issues which should be filtered.</param>
+        /// <returns>List of issues filtered to only the ones affecting files changed in this pull request.</returns>
+        private IList<ICodeAnalysisIssue> FilterIssuesByPath(IList<ICodeAnalysisIssue> issues)
+        {
+            if (!issues.Any())
+            {
+                return issues;
+            }
+
+            var modifiedFilesInPullRequest = this.pullRequestSystem.GetModifiedFilesInPullRequest().ToList();
+            this.log.Verbose(
+                "Files changed in this pull request:\n{0}",
+                string.Join(
+                    Environment.NewLine,
+                    modifiedFilesInPullRequest.Select(x => "  " + x)));
+
+            var countBefore = issues.Count;
+            var result =
+                issues
+                    .Where(issue => modifiedFilesInPullRequest.Select(x => x.ToString()).Contains(issue.AffectedFileRelativePath.ToString()))
```

Comment:
Changed it to do the transformation once in the beginning and changed local variable to a HashSet